gammapy/analysis/config.py

```diff
@@ -66,127 +79,212 @@ class Config:
     level: str = "info"
     filename: Path = None
     filemode: str = None
     format: str = None
     datefmt: str = None
 
 
-class General(GammapyBaseModel):
-    log: Log = Log()
+class GeneralConfig(GammapyBaseConfig):
+    log: LogConfig = LogConfig()
     outdir: str = "."
 
 
-class AnalysisConfig(GammapyBaseModel):
+class AnalysisConfig(GammapyBaseConfig):
     """Config class handling the high-level interface settings."""
 
-    general: General = General()
-    data: Data = Data()
-    datasets: Datasets = Datasets()
-    fit: Fit = Fit()
-    flux_points: FluxPoints = FluxPoints()
+    general: GeneralConfig = GeneralConfig()
+    observations: ObservationsConfig = ObservationsConfig()
+    datasets: DatasetsConfig = DatasetsConfig()
+    fit: FitConfig = FitConfig()
+    flux_points: FluxPointsConfig = FluxPointsConfig()
+
+    @classmethod
+    def from_template(cls, template):
+        """Create AnalysisConfig from existing templates.
+
+        Parameters
+        ----------
+        template : {"1d", "3d"}
+            Built-in templates.
+
+        Returns
+        -------
+        config : `AnalysisConfig`
+            AnalysisConfig class
+        """
+        filename = CONFIG_PATH / ANALYSIS_TEMPLATES[template]
+        return cls.read(filename)
+
+    def __str__(self):
+        """Display settings in pretty YAML format."""
+        info = self.__class__.__name__ + "\n\n\t"
+
+        data = self.to_yaml()
+        data = data.replace("\n", "\n\t")
+        info += data
+        return info.expandtabs(tabsize=4)
+
+    @classmethod
+    def read(cls, filename):
+        config = read_yaml(filename)
+        return cls.from_yaml(config)
+
+    @classmethod
+    def from_yaml(cls, settings):
+        if isinstance(settings, str):
+            settings = yaml.safe_load(settings)
+        return AnalysisConfig(**settings)
+
+    def write(self, filename, overwrite=False):
+        fname = Path(filename).name
+        fpath = Path(self.general.outdir) / fname
+        if fpath.exists() and not overwrite:
+            raise IOError(f"File {filename} already exists.")
+        fpath.write_text(self.to_yaml())
+        log.info(f"Configuration settings saved into {fpath}")
+
+    def to_yaml(self):
+        return yaml.dump(yaml.safe_load(self.json()), sort_keys=False, indent=4)
+
+    def set_logging(self):
+        """Set logging parameters for API."""
+        self.general.log.level = self.general.log.level.upper()
+        logging.basicConfig(**self.general.log.dict())
+        log.info("Setting logging config: {!r}".format(self.general.log.dict()))
+
+    def update(self, config=None, filename=""):
+        """Updates config with provided settings.
+
+         Parameters
+         ----------
+         config : string dict or `AnalysisConfig` object
+             Configuration settings provided in dict() syntax.
+         filename : string
+             Filename in YAML format.
+         """
+        try:
```

Comment:
Remove the complex try - except and I/O here.
A user can just write `config.update(yaml.load("config.yaml"))` if they want to read config from YAML and update an existing config, no?

Reply:
Done